File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Library.py
from typing import Optional, List
from pydantic import *
from ..base.UsageContext import UsageContext
from ..base.Attachment import Attachment
from ..base.Period import Period
from ..base.ContactDetail import ContactDetail
from ..base.DataRequirement import DataRequirement
from ..base.CodeableConcept import CodeableConcept
from ..base.RelatedArtifact import RelatedArtifact
from ..base.DomainResource import DomainResource
from ..base.ParameterDefinition import ParameterDefinition
from ..base.Reference import Reference
from ..base.Identifier import Identifier

class Library(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    data_requirement: Optional[List[DataRequirement]] = None
    endorser: Optional[List[ContactDetail]] = None
    publisher: Optional[str] = None
    approval_date: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    purpose: Optional[str] = None
    content: Optional[List[Attachment]] = None
    subject_codeable_concept: Optional[CodeableConcept] = None
    name: Optional[str] = None
    use_context: Optional[List[UsageContext]] = None
    copyright: Optional[str] = None
    type: CodeableConcept
    experimental: Optional[bool] = None
    topic: Optional[List[CodeableConcept]] = None
    title: Optional[str] = None
    author: Optional[List[ContactDetail]] = None
    usage: Optional[str] = None
    status: str
    subtitle: Optional[str] = None
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    last_review_date: Optional[str] = None
    editor: Optional[List[ContactDetail]] = None
    reviewer: Optional[List[ContactDetail]] = None
    version: Optional[str] = None
    related_artifact: Optional[List[RelatedArtifact]] = None
    contact: Optional[List[ContactDetail]] = None
    subject_reference: Optional[Reference] = None
    parameter: Optional[List[ParameterDefinition]] = None
    effective_period: Optional[Period] = None

File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/Library.ts
import { UsageContext } from "./UsageContext";
import { Attachment } from "./Attachment";
import { Period } from "./Period";
import { ContactDetail } from "./ContactDetail";
import { DataRequirement } from "./DataRequirement";
import { CodeableConcept } from "./CodeableConcept";
import { RelatedArtifact } from "./RelatedArtifact";
import { DomainResource } from "./DomainResource";
import { ParameterDefinition } from "./ParameterDefinition";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";

export type Library = DomainResource & {
    description?: string;
    date?: string;
    dataRequirement?: DataRequirement[];
    endorser?: ContactDetail[];
    publisher?: string;
    approvalDate?: string;
    jurisdiction?: CodeableConcept[];
    purpose?: string;
    content?: Attachment[];
    name?: string;
    useContext?: UsageContext[];
    copyright?: string;
    type: CodeableConcept;
    experimental?: boolean;
    topic?: CodeableConcept[];
    title?: string;
    author?: ContactDetail[];
    usage?: string;
    status: string;
    subtitle?: string;
    url?: string;
    identifier?: Identifier[];
    lastReviewDate?: string;
    editor?: ContactDetail[];
    reviewer?: ContactDetail[];
    version?: string;
    relatedArtifact?: RelatedArtifact[];
    contact?: ContactDetail[];
    subject?: CodeableConcept | Reference;
    parameter?: ParameterDefinition[];
    effectivePeriod?: Period;
};